python/trifinger_simulation/trifinger_platform.py

```diff
@@ -38,15 +39,16 @@ def __init__(self):
         self.timestamp = None
 
 
-class TriCameraObservation:
-    """Pure-python copy of trifinger_cameras.tricamera.TriCameraObservation."""
+class TriCameraObjectObservation:
```

Comment:
I am a bit confused by this design. In my opinion, it would be better to have the object pose separate from the camera observation. From what I understand, the object pose is actually as returned by the pybullet simulator, so the ground truth, and it is not inferred by the camera observations. Also, even if it is, I would rather it be separate from the camera observation so that it more modular that way. So, later, when we add support to get the object pose from the tracker, we can have an option in this object observation to either get the pose estimate from pybullet, or from the tracker. Like in any case, it wouldn't be the camera observation per se but rather the output of a different method (feature) that will access the camera observation and return the object pose from that, so I think it shouldn't be a part of the camera observation.

Reply:
I agree that it is not the best design but it is implemented like this for the real robot (it makes it much easier to handle in our current setup to have the object tracking running directly in the camera driver).  So the change here is just replicating the API of the real system.

Reply:
Okay. :+1: